backstory/src/test/java/com/google/sps/servlets/BlobstoreManagerTest.java

```diff
@@ -0,0 +1,114 @@
+import com.google.appengine.api.blobstore.BlobKey;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * Tests for the blobstore service wrapper, BlobstoreManager
+ */
+@RunWith(MockitoJUnitRunner.class)
+public final class BlobstoreManagerTest {
+  /**
+   * Tests the getUploadedFileBlobKeyString method of BlobstoreManager
+   */
+  @Test
+  public void testGetUploadedFileBlobKeyString() {
+    // Creates the required mocks for an HTTP request/response.
+    HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+
+    BlobstoreService mockBlobstoreService = mock(BlobstoreService.class);
+    BlobstoreServiceConstantFields mockBlobstoreServiceConstantFields = mock(BlobstoreServiceConstantFields.class);
+    BlobInfoFactory mockBlobInfoFactory = mock(BlobInfoFactory.class);
+    BlobInfo mockBlobInfo = mock(BlobInfo.class);
+
+    String formInputElementName = "image-upload";
+    String mockBlobkey = "mockBlobkey";
+    Map<String, List<BlobKey>> mockInputToBlobkey = new HashMap<>();
+    List<BlobKey> mockBlobKeys = new ArrayList<>();
+    mockBlobKeys.add(new BlobKey(mockBlobkey));
+    mockInputToBlobkey.put(formInputElementName, mockBlobKeys);
+
+    when(mockBlobstoreService.getUploads(any(HttpServletRequest.class)))
+        .thenReturn(mockInputToBlobkey);
+
+    // Mock one image getting uploaded
+    when(mockBlobInfo.getSize()).thenReturn(new Long(1));
+    when(mockBlobInfoFactory.loadBlobInfo(any(BlobKey.class)))
+        .thenReturn(mockBlobInfo);
+
+    BlobstoreManager blobstoreManager = new BlobstoreManager(mockBlobstoreService, mockBlobstoreServiceConstantFields, mockBlobInfoFactory);
+    Assert.assertEquals(mockBlobkey, blobstoreManager.getUploadedFileBlobKeyString(mockRequest, formInputElementName));
+  }
+
+  /**
+   * Tests the getBlobBytes method of BlobstoreManager
+   */
+  @Test
+  public void testGetBlobBytes() throws IOException {
+    // Creates the required mocks for an HTTP request/response.
+    HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+
+    BlobstoreService mockBlobstoreService = mock(BlobstoreService.class);
+    BlobstoreServiceConstantFields mockBlobstoreServiceConstantFields = mock(BlobstoreServiceConstantFields.class);
+    BlobInfoFactory mockBlobInfoFactory = mock(BlobInfoFactory.class);
+    BlobInfo mockBlobInfo = mock(BlobInfo.class);
+
+    byte[] mockImageByteArray = new byte[0];
+    // Mock a max blob fetch size of 1
+    when(mockBlobstoreServiceConstantFields.getMaxBlobFetchSize())
+        .thenReturn(1);
+    when(mockBlobstoreService.fetchData(
+      any(BlobKey.class), anyLong(), anyLong()
+    )).thenReturn(mockImageByteArray);
+
+    String formInputElementName = "image-upload";
+    String mockBlobkey = "mockBlobkey";
```

Comment:
Some of this mocking seems very repetitive from the last test. Could you maybe consolidate it (as much as possible) in a helper method?